docs/process.md

```diff
@@ -0,0 +1,41 @@
+# Development Process
+
+Notes:
+* The development process is a simplified version of the one used in [TEAMMATES](https://github.com/TEAMMATES/repo/blob/master/devdocs/process.md). 
+* The `master` branch always contains the latest stable code (may contain unreleased features/fixes).
+* The `release` branch always contains the latest released version.
```

Comment:
I think we need another branch (`rc`, as in release candidate?). Currently, the documentation users see is one week ahead of the version they use because we have a one week dog fooding period.